CoreLayersView/src/au/gov/asd/tac/constellation/views/layers/LayersViewController.java

```diff
@@ -67,36 +70,27 @@ public void edit(final GraphWriteMethods graph, final PluginInteraction interact
     }
 
     /**
+     * Get all layer queries from the Layer View and store them on the qraph.
      * Update the bitmask used to determine visibility of elements on the graph.
      */
     public void execute() {
         final LayersViewPane pane = parent.getContent();
         if (pane == null) {
             return;
         }
+        final List<String> layerQueries = new ArrayList<>();
         int newBitmask = 0b0;
         for (final LayerDescription layer : pane.getlayers()) {
+            layerQueries.add(layer.getLayerQuery().isEmpty() ? null : layer.getLayerQuery());
             newBitmask |= layer.getCurrentLayerVisibility() ? (1 << layer.getLayerIndex() - 1) : 0;
         }
         // if the newBitmask is 1, it means none of the boxes are checked. therefore display default layer 1 (All nodes)
         newBitmask = (newBitmask == 0) ? 0b1 : (newBitmask > 1) ? newBitmask & ~0b1 : newBitmask;
-        PluginExecution.withPlugin(new UpdateGraphBitmaskPlugin(newBitmask))
+
+        Future<?> layerQueryFuture = PluginExecution.withPlugin(new UpdateGraphQueriesPlugin(layerQueries))
                 .executeLater(GraphManager.getDefault().getActiveGraph());
```

Comment:
Is the active graph required here? Just wondering about executing this all standalone (e.g. in a unit test)

Reply:
I think I get where you are coming from - in the case where there is no active graph, or the graph is closed? 
I'm not too aware of other ways - I've seen this way as a standard throughout the project - I could be using it in the wrong context though. @arcturus2 Would a null check here or within the plugin to be executed be necessary?

Reply:
> I've seen this way as a standard throughout the project - I could be using it in the wrong context though.

I agree @aldebaran30701 it used throughout the code base and context does matter. In general a view should be fine to make the `getActiveGraph()` call though checking for null here would be required.

I've opted to not do a null check in `executeLater` deliberately because situations like this can go unnoticed.

Reply:
In an attempt to not let situations like this go unnoticed should we add a warning log or something if the current active graph is null?
If the current active graph is null, nothing further should be done correct?

Reply:
Yeah @aldebaran30701 I think logging it is a great idea.

So on the topic of things silently passing/failing I commented https://github.com/constellation-app/constellation/pull/602#discussion_r447458005 that it turns out the plugin is not run in the unit test; a good example of how things get missed. Doing the null check before running `executeLater` to my mind makes it more obvious when the plugin runs.

Reply:
>  If the current active graph is null, nothing further should be done correct?

Correct. In `executeNow()` it looks like there is a check for `null` but not in `executeLater`. One silently fails and the other throws an exception!

Perhaps a `null` check and logging it would be fine. This would be a change in the `DefaultPluginEnvironment` etc and a separate PR I guess. Want me to do it or are you ok to @aldebaran30701 ?

Reply:
Just to reiterate - This code block does not change, but a separate Pr which has a null check and error logging within `DefaultPluginEnvironment ` is required correct?

Reply:
Yeah @aldebaran30701 and this is largely to keep your PR focused on the layers code and also because the plugin framework not dealing with nulls (on purpose) is a different problem.